infra/src/test/kotlin/fi/fta/geoviite/infra/linking/LinkingDaoIT.kt

```diff
@@ -26,66 +20,60 @@ import org.springframework.test.context.ActiveProfiles
 @SpringBootTest
 class LinkingDaoIT @Autowired constructor(
     private val linkingDao: LinkingDao,
-    private val switchService: LayoutSwitchService,
-    private val locationTrackService: LocationTrackService,
 ) : DBTestBase() {
 
     @Test
     fun noSwitchBoundsAreFoundWhenNotLinkedToTracks() {
-        val switch = switchService.getOrThrow(
-            MainLayoutContext.draft,
-            switchService.saveDraft(LayoutBranch.main, switch(1, draft = true)).id,
-        )
-        assertEquals(null, linkingDao.getSwitchBoundsFromTracks(MainLayoutContext.official, switch.id as IntId))
-        assertEquals(null, linkingDao.getSwitchBoundsFromTracks(MainLayoutContext.draft, switch.id as IntId))
+        val switchId = mainDraftContext.insertSwitch().id
+        assertEquals(null, linkingDao.getSwitchBoundsFromTracks(MainLayoutContext.official, switchId))
+        assertEquals(null, linkingDao.getSwitchBoundsFromTracks(MainLayoutContext.draft, switchId))
     }
 
     @Test
     fun switchBoundsAreFoundFromTracks() {
-        val trackNumber = getOrCreateTrackNumber(TrackNumber("123"))
-        val tnId = trackNumber.id as IntId
-        val switch = switchService.getOrThrow(
-            MainLayoutContext.draft,
-            switchService.saveDraft(LayoutBranch.main, switch(1, draft = true)).id,
-        )
+        val tnId = mainDraftContext.insertTrackNumber().id
+        val switchId = mainDraftContext.insert(switch(1)).id
 
         val point1 = Point(10.0, 10.0)
         val point2 = Point(12.0, 10.0)
         val point3_1 = Point(10.0, 12.0)
         val point3_2 = Point(10.0, 13.0)
 
         // Linked from the start only -> second point shouldn't matter
-        locationTrackService.saveDraft(
-            LayoutBranch.main,
-            locationTrack(tnId, externalId = someOid(), draft = true).copy(
```

Comment:
Näitä `copy()` -juttuja on muuten aika monessa paikassa myös ihan turhaan historian jäännöksenä. Samat parametrit voi antaa suoraan rakennusfunkkariin.